index.html

```diff
@@ -1107,9 +1460,80 @@ <h2>Reporting of Third-party Subresource Fetch Failures</h2>
     <section>
       <h2>Privacy Considerations</h2>
 
-      <p>NEL provides network error reports that could expose new information about the user's network configuration. For example, an attacker could abuse NEL reporting to probe users network configuration. Also, similar to HSTS, HPKP, and pinned CSP policies, the stored <a>NEL policy</a> could be used as a "supercookie" by setting a distinct policy with a custom (per-user) reporting URI to act as an identifier in combination with (or instead of) HTTP cookies.</p>
+      <p>
+      NEL provides network error reports that could expose new information about
+      the user's network configuration. For example, an attacker could abuse NEL
+      reporting to probe the user's network configuration, or to scan for
+      servers on the user's internal network. Also, similar to HSTS, HPKP, and
+      pinned CSP policies, the stored <a>NEL policy</a> could be used as a
+      "supercookie" by setting a distinct policy with a custom (per-user)
+      reporting URI to act as an identifier in combination with (or instead of)
+      HTTP cookies.
+      </p>
+
+      <p>
+      To mitigate some of the above risks, NEL registration is restricted to
+      <a>trustworthy origins</a>, and delivery of network error reports is
+      similarly restricted to <a>trustworthy origins</a>. This disallows a
+      transient HTTP MITM from trivially abusing NEL as a persistent
+      tracker.

```

Comment:
IIUC, that means that NEL policies can effectively act as third party cookies, when delivered on third party responses. Is that correct?

If so, might be worth while to mention that and indicate that:
* The policy cache should be purged when cookies are purged
* Third party policies should be treated as third party cookies (and e.g. not be saved when the user preferences prevent third party cookies)

Reply:
I think a NEL policy can act as a first-party cookie, but not as a third-party one.  (Unless I'm misunderstanding the threat you're suggesting.)  NEL policies can only be set by the origin that the requests will be sent to; this ¶ describes how we try to enforce that.  i.e. a NEL policy can't be delivered on a third-party response — the policy doesn't contain an explicit origin describing what it should apply to, it implicitly applies to the origin of the response itself.

Re clearing the policy cache, there's an existing ¶ at the bottom of the section that requires that.